File: 프로그래머스/2/42586. 기능개발/기능개발.kt
import kotlin.math.ceil

class Solution {
fun solution(progresses: IntArray, speeds: IntArray): IntArray {
    val deployDays = mutableListOf<Int>()
    for (i in progresses.indices) {
        val remain = 100 - progresses[i]
        val day = (remain + speeds[i] - 1) / speeds[i]   // 올림 나눗셈
        deployDays.add(day)
    }

    val result = mutableListOf<Int>()
    var previous = deployDays[0]
    var cnt = 1

    for (i in 1 until deployDays.size) {
        val day = deployDays[i]
        if (day <= previous) {
            cnt++
        } else {
            result.add(cnt)
            cnt = 1
            previous = day
        }
    }
    result.add(cnt) // 마지막 묶음 추가

    return result.toIntArray()
}
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    finishdays = []
    

    for i in range(len(progresses)):
        day = math.ceil((100 - progresses[i]) / speeds[i])
        finishdays.append(day)

    cnt = 0
    previous = finishdays[0]

    for d in finishdays:
        if(d <= previous):
            cnt += 1
        else :
            answer.append(cnt)
            cnt = 1
            previous = d

    answer.append(cnt)

    return answer

